### PR TITLE
add contracts store

### PR DESCRIPTION
**Pull request by upacyxou**
- contracts store added
- contract syncrhonizer added
- `getOwnedEERC721` function added
- additional utils and models added

### CODE REVIEW

File: src/ContractsStore.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+} from './helpers/ContractSynchronizer'
+import Network from './models/Network'
+import transformObjectValues from './helpers/transformObjectValues'
+
+export class ContractsStore extends PersistableStore {
+  connectedAccounts: { [account: string]: ContractSynchronizer } = {}
+  currentBlock?: number
+  addressToTokenIds?: Promise<{ [address: string]: string[] } | undefined>
+
+  getLogs: (filter: providers.Filter) => Promise<providers.Log[]>
```
Comment (T-Damer) [inline src/ContractsStore.ts]: Can we import this stuff like this:
```TS
import { Log, Filter, etc } from '@ethersproject/providers'
```
?

File: src/ContractsStore.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  get persistanceName() {
+    return `${this.constructor.name}_${this.network}`
+  }
+
+  provider: providers.Provider
```
Comment (T-Damer) [inline src/ContractsStore.ts]: Let's import this type from `'@ethersproject/providers'`

File: src/ContractsStore.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  fetchBlockNumber() {
+    return this.provider.getBlockNumber()
+  }
+
+  async fetchMoreContractsOwned(
+    account: string,
+    accountChange?: boolean,
+    startBlock?: number
+  ) {
+    if (!this.currentBlock) this.currentBlock = await this.fetchBlockNumber()
```
Comment (T-Damer) [inline src/ContractsStore.ts]: Let's wrap this one into `{}`, it's hard to read without them

File: src/ContractsStore.ts (hunk 1 above)
Comment (upacyxou) [reply src/ContractsStore.ts]: yes, I'll fix that

**Review by T-Damer — APPROVED** [review]
lgtm

**Pull request merged**